TypeCobol.Test/Parser/Incremental/TestIncrementalPipeline.cs

```diff
@@ -1,95 +1,61 @@
-            TestUnit unit = new TestUnit(new Multipass(paths));
-            unit.Parse();
+            var unitTest = new TypeCobol.Test.UtilsNew.TestUnit(Path.Combine(Root, "Simple.pgm"), DocumentFormat.FreeUTF8Format);
 
-            var e = UpdateLine(TextChangeType.LineInserted, 2, "END PROGRAM Simple.");
-            e = UpdateLine(TextChangeType.LineUpdated, 1, "PROGRAM-ID. Simpler.", e);
+            var inputChangesFilePath = Path.Combine(Root, "Simple_EndProgramAddProgramId.inc");
+            foreach (var inputChange in TypeCobol.Test.UtilsNew.InputChange.Load(inputChangesFilePath))
+            {
+                unitTest.AddInputChange(inputChange);
+            }
 
-            // clear document
-            unit.Compiler.CompilationResultsForProgram.UpdateTextLines(e);
-            unit.Parse();
+            var formatter = new TypeCobol.Test.UtilsNew.CodeElements();
+            var first = Path.Combine(Root, "Simple.1.txt");
+            unitTest.AddComparison(new TypeCobol.Test.UtilsNew.Comparison("1", first, formatter));
+            var second = Path.Combine(Root, "Simple.2.txt");
+            unitTest.AddComparison(new TypeCobol.Test.UtilsNew.Comparison("2", second, formatter));
 
-            var names = unit.Comparator.paths.Resultnames as Multipass.IndexNames;
-            if (names != null) names.index = 2;
-            Console.WriteLine("Compare with result file: " + unit.Comparator.paths.Result);
-            unit.Compare();//with Simple.2.txt
+            unitTest.Run();
         }
 
         [TestMethod]
         [TestCategory("Incremental")]
         [TestProperty("Time", "long")]
         public void Check_BeforeAfterInsertion()
         {
-            Paths paths = new Paths(Root, Root, Root + Path.DirectorySeparatorChar + "Simple.pgm", new Multipass.IndexNames());
-            TestUnit unit = new TestUnit(new Multipass(paths));
-            unit.Parse();
-            var names = unit.Comparator.paths.Resultnames as Multipass.IndexNames;
-            if (names != null)
-            {
-                names.index = 0;
-                unit.Compare();//with Simple.0.txt
-
-                // explicitely close program by adding END PROGRAM line
-                var e2 = UpdateLine(TextChangeType.LineInserted, 2, "END PROGRAM Simple.");
-                var e = UpdateLine(TextChangeType.LineUpdated, 1, "  PROGRAM-ID. Simple.");
-                e.TextChanges.Add(e2.TextChanges[0]);
-                unit.Compiler.CompilationResultsForProgram.UpdateTextLines(e);
-                unit.Parse();
-                names.index++;
-                Console.WriteLine("Compare with result file: " + unit.Comparator.paths.Result);
-                unit.Compare();//with Simple.1.txt
+            var unitTest = new TypeCobol.Test.UtilsNew.TestUnit(Path.Combine(Root, "Simple.pgm"), DocumentFormat.FreeUTF8Format);
```

Comment:
As discussed, I'd prefer these tests to be done with generic files like the ones already in Incremental folder.
So it would be:
```
Simple_EndProgramAddProgramId.pgm
Simple_EndProgramAddProgramId.inc
Simple_EndProgramAddProgramId.0.PGM.txt
Simple_EndProgramAddProgramId.1.PGM.txt
Simple_EndProgramAddProgramId.2.PGM.txt
Simple_EndProgramAddProgramId.3.PGM.txt

Simple_EndProgramAddProgramIdClear.pgm
Simple_EndProgramAddProgramIdClear.inc
Simple_EndProgramAddProgramIdClear.0.PGM.txt
Simple_EndProgramAddProgramIdClear.1.PGM.txt
Simple_EndProgramAddProgramIdClear.2.PGM.txt
Simple_EndProgramAddProgramIdClear.3.PGM.txt
```
It's easier to read and can be understood without reading the C# code.

Reply:
Done. I've kept the CodeElement format to match the existing test. This should be still readable though.